test/infra_test.go

```diff
@@ -1,126 +1,229 @@
+	authinfos[namespace] = &clientcmdapi.AuthInfo{
+		Token: tok.Token,
+	}
 
-		if strings.Contains(service.Name, filter) {
-			serviceName := service.Name
-			return serviceName
-		}
+	clientConfig := clientcmdapi.Config{
+		Kind:           "Config",
+		APIVersion:     "v1",
+		Clusters:       clusters,
+		Contexts:       contexts,
+		CurrentContext: *cluster.Arn,
+		AuthInfos:      authinfos,
 	}
-	return serviceName
+
+	clientcmd.WriteToFile(clientConfig, RecommendedHomeFile)
+
+	return clientset, nil
 }
 
-func verifyPrometheusWelcomePage(statusCode int, body string) bool {
-	if statusCode != 200 {
-		return false
+// Returns running pods
+func getRunningPods(clientset *kubernetes.Clientset, result *eks.DescribeClusterOutput) *corev1.PodList {
+
+	// Use clientset to interact with k8s cluster
+	// https://github.com/kubernetes/client-go/blob/master/examples/out-of-cluster-client-configuration/main.go#L64
+	pods, err := clientset.CoreV1().Pods(namespace).List(context.TODO(), v1.ListOptions{FieldSelector: "status.phase=Running"})
+	if err != nil {
+		log.Fatalf("Error getting EKS pods: %v", err)
+	}
+	log.Printf("There are %d pods associated with cluster: %s", len(pods.Items), *result.Cluster.Name)
+
+	return pods
+}
+
+// Wait services
+func waitServices(t *testing.T, clientset *kubernetes.Clientset, result *eks.DescribeClusterOutput) {
+
+	svcs, err := clientset.CoreV1().Services(namespace).List(context.TODO(), v1.ListOptions{FieldSelector: "metadata.namespace=" + namespace})
+	if err != nil {
+		log.Fatalf("Error getting EKS services: %v", err)
+	}
+	log.Printf("There are %d services associated with cluster: %s", len(svcs.Items), *result.Cluster.Name)
+
+	for i := range svcs.Items {
+
+		// Wait all deployed services to be available and ready
+		k8s.WaitUntilServiceAvailable(t, kubeOptions, svcs.Items[i].Name, 60, 1*time.Second)
+
 	}
 
-	return strings.Contains(body, `<title>Prometheus Time Series Collection and Processing Server</title>`)
 }
 
-func verifyGrafanaWelcomePage(statusCode int, body string) bool {
+func verifyPrometheus(statusCode int, body string) bool {
 	if statusCode != 200 {
 		return false
 	}
 
-	return strings.Contains(body, `<title>Grafana</title>`)
+	return strings.Contains(body, "Prometheus Time Series Collection and Processing Server")
 }
 
-func TestingPrometheusDeployment(t *testing.T) {
+func ValidatePrometheusDeployment(t *testing.T, clientset *kubernetes.Clientset, result *eks.DescribeClusterOutput) {
 
-	var filter string = "server"
+	pods := getRunningPods(clientset, result)
 
-	prometheusPod := awaitPods(t, kubeOptions, filter)
+	for i := range pods.Items {
 
-	awaitServices(t, kubeOptions, filter)
+		// Wait all deployed pods to be available and ready
+		k8s.WaitUntilPodAvailable(t, kubeOptions, pods.Items[i].Name, 60, 1*time.Second)
 
-	tunnel := k8s.NewTunnel(kubeOptions, k8s.ResourceTypePod, prometheusPod, 0, 9090)
-	defer tunnel.Close()
-	tunnel.ForwardPort(t)
+		var filter string = "server"
```

Comment:
use short assignment instead

Reply:
Actually you can use "server" directly no need to use in a variable at all, or move to const